closure/goog/graphics/canvasgraphics_test.js

```diff
@@ -0,0 +1,73 @@
+// Copyright 2015 The Closure Library Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+goog.provide('goog.graphics.CanvasGraphicsTest');
+goog.setTestOnly('goog.graphics.CanvasGraphicsTest');
+
+goog.require('goog.dom');
+goog.require('goog.graphics.CanvasGraphics');
+goog.require('goog.graphics.SolidFill');
+goog.require('goog.graphics.Stroke');
+goog.require('goog.testing.jsunit');
+
+
+var graphics;
+
+function setUp() {
+  graphics = new goog.graphics.CanvasGraphics(100, 100);
+  graphics.createDom();
+  if (!graphics.canvas_.getContext) {
+    // no browser support for canvas
+    graphics = null;
+    return;
+  }
```

Comment:
There is actually a `shouldRunTests` method specifically designed for this: https://github.com/google/closure-library/blob/master/closure/goog/testing/testcase.js#L417